core/store/config/general_config.go

```diff
@@ -286,6 +289,11 @@ func (c *generalConfig) SetDB(db *gorm.DB) {
 	c.ORM = orm
 }
 
+// SetKeyStore provides reference to the keystore for runtime configuration values
```

Comment:
Seems like something that should be passed in the constructor of *generalConfig right (same goes for SetDB actually)? I don't think we want to be able to overwrite this dynamically ?

Reply:
I agree, I think there's a bit of a chicken and egg problem though, since keystore takes the config object in it's constructor. There is probably a larger-scale refactoring of config construction / dependencies that is in order. (But I was hoping to avoid in this PR).